src/colmap/estimators/bundle_adjustment.h

```diff
@@ -74,7 +74,7 @@ struct BundleAdjustmentOptions {
 
   BundleAdjustmentOptions() {
     solver_options.function_tolerance = 0.0;
-    solver_options.gradient_tolerance = 0.0;
+    solver_options.gradient_tolerance = 1e-4;
```

Comment:
The ceres default is 1e-10. colmap default for incremental global BA is 1.0 and for local BA is 10.0. For custom global BA convergence, the 1e-4 should be a conservative value. I tested this on a few scenes and it converges before reaching the full iterations.